File: WorkerManager/app/src/main/java/com/example/workermanager/workmanager/DownloadingWorker.java
package com.example.workermanager.workmanager;

import android.annotation.SuppressLint;
import android.content.Context;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

import java.text.SimpleDateFormat;
import java.util.Date;

public class DownloadingWorker extends Worker {
    public DownloadingWorker(@NonNull Context context, @NonNull WorkerParameters workerParams) {
        super(context, workerParams);
    }

    @NonNull
    @Override
    public Result doWork() {
        try {
            for (int i=0; i<4000; i++) {
                Log.i("MyTag", "DownloadWorker" + i);
            }
                @SuppressLint("SimpleDateFormat")
                SimpleDateFormat time = new SimpleDateFormat("dd/M/yyyy hh:mm:ss");
                String currentDate = time.format(new Date());
                Log.i("MyTag",currentDate);
               return Result.success();
        }catch (Throwable t){
            return Result.failure();
        }

    }
}

File: WorkerThread/app/src/main/java/com/example/workerthread/DownloadingWorker.kt
package com.example.workerthread

import android.content.Context
import android.util.Log
import androidx.work.Worker
import androidx.work.WorkerParameters
import java.lang.Exception
import java.text.SimpleDateFormat
import java.util.*

class DownloadingWorker (context : Context, params:WorkerParameters) : Worker(context,params){
    override fun doWork(): Result {
        return try {
            for (i in 0..4000){
                Log.i("MyTag","FilterWorker $i")
            }
            val time = SimpleDateFormat("dd/M/yyyy hh:mm:ss")
            val currentDate = time.format(Date())
            Log.i("MyTag",currentDate)
            Result.success()
        }catch (e:Exception){
            Result.failure()
        }
    }

}